### PR TITLE
Playlist item modal

### PR DESCRIPTION
**Pull request by ezequielc**
@alex-deaconu @rodrigopavezi please review. Thanks!

### CODE REVIEW

**Comment by rodrigopavezi** [comment]
@ezequielc LG :+1:

File: js/services/svc-placeholder-playlist-factory.js [hunk 1]
```diff
@@ -26,6 +32,23 @@ angular.module('risevision.editorApp.services')
         }
       };
 
+      factory.getCurrentItemProperties = function() {
+        if (!factory.item) {
+          return undefined;
+        }
+        var cloneItem = {};
+        for(var i = 0; i < PLAYLIST_ITEM_MODAL_FIELDS.length; i++){
+          cloneItem[PLAYLIST_ITEM_MODAL_FIELDS[i]] = factory.item[PLAYLIST_ITEM_MODAL_FIELDS[i]];
```
Comment (alex-deaconu) [inline js/services/svc-placeholder-playlist-factory.js]: @ezequielc why are you iterating through all the fields instead of using `angular.copy`? Thanks.

Comment (ezequielc) [inline js/services/svc-placeholder-playlist-factory.js]: @alex-deaconu there are some fields we are not using in the modal like objectData and additionalParams, so I preferred to leave them untouched.

Comment (alex-deaconu) [inline js/services/svc-placeholder-playlist-factory.js]: @ezequielc actually, we will need objectData and additionalParams in the modal for editing the Widget settings and retrieving the Widget name, url, etc (via objectData). Thanks.

Comment (ezequielc) [inline js/services/svc-placeholder-playlist-factory.js]: ok @alex-deaconu I will change to `angular.copy` then. And for setting the values back, would you recommend iterating or setting one by one?

Comment (alex-deaconu) [inline js/services/svc-placeholder-playlist-factory.js]: @ezequielc you can take a look here: https://github.com/Rise-Vision/schedules-app/blob/master/js/controllers/ctr-playlist-item-modal.js. We use `angular.copy` for both duplicating the item, and updating it. Thanks.

File: js/controllers/ctr-playlist-item-modal.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+'use strict';
+
+angular.module('risevision.editorApp.controllers')
+  .controller('PlaylistItemModalController', ['$scope',
+    '$modalInstance', 'placeholderPlaylistFactory',
+    function ($scope, $modalInstance, placeholderPlaylistFactory) {
+
+      $scope.item = placeholderPlaylistFactory.getCurrentItemProperties();
```
Comment (alex-deaconu) [inline js/controllers/ctr-playlist-item-modal.js]: @ezequielc I'm wondering if we should use the same pattern as here: https://github.com/Rise-Vision/schedules-app/blob/master/js/controllers/ctr-playlist-item-modal.js, and pass the playlistItem to the modal via `resolve` to the controller as an injection. And skip the factory all together for managing the item. Thanks.

Comment (ezequielc) [inline js/controllers/ctr-playlist-item-modal.js]: @alex-deaconu I liked the suggestion. I will implement it. Thanks.

**Comment by ezequielc** [comment]
@alex-deaconu I implemented the suggested changes. Please review. Thanks!

**Comment by alex-deaconu** [comment]
@ezequielc thanks for updating :+1: .

**Pull request merged**